src/index.js

```diff
@@ -38,6 +36,17 @@ export default function nodeResolve ( options = {} ) {
 			// disregard entry module
 			if ( !importer ) return null;
 
+			if (options.browser && browserMapCache[importer]) {
+				const browser = browserMapCache[importer];
+				if (browser[importee]) {
+					importee = browser[importee];
+				}
+				if (browser[importee] === false) {
+					return ES6_BROWSER_EMPTY;
+				}
+			}
```

Comment:
This is the crux of the replacement logic with `browser-resolve`. Given a map from an `importer` - resolve to the values of that key. Here we can also take advantage of returning `ES6_BROWSER_EMPTY` as early as possible - as we don't need to resolbe that beyond what it is doing.